tests-e2e/fixture/application/fixture.go

```diff
@@ -83,6 +83,36 @@ func HaveAutomatedSyncPolicy(syncPolicy appv1alpha1.SyncPolicyAutomated) matcher
 	}, BeTrue())
 }
 
+func HaveSyncOption(syncOption appv1alpha1.SyncOptions) matcher.GomegaMatcher {
+
+	return WithTransform(func(app appv1alpha1.Application) bool {
+
+		config, err := fixture.GetServiceProviderWorkspaceKubeConfig()
+		Expect(err).To(BeNil())
+
+		k8sClient, err := fixture.GetKubeClient(config)
+		if err != nil {
+			fmt.Println(k8sFixture.K8sClientError, err)
+			return false
+		}
+
+		err = k8sClient.Get(context.Background(), client.ObjectKeyFromObject(&app), &app)
+		if err != nil {
+			fmt.Println(k8sFixture.K8sClientError, err)
+			return false
+		}
+
+		fmt.Println(app.Spec.SyncPolicy.SyncOptions)
+		isSyncOption := false
```

Comment:
The passed-in `syncOption` should never be nil, and that's what should be tested against instead of the hard-coded `CreateNamespace=true`.  Currently the actual value of `syncOption` is ignored.

I think the type of the parameter to this function should be changed to a string, and it should test that the application has that specified option.

Reply:
What I think is that this test is not checking whether the syncOption exists or not but it checks whether the syncOption has a particular option (CreateNamespace=true) and that is done in this block :

```
if syncOption != nil {
			isSyncOption = app.Spec.SyncPolicy.SyncOptions.HasOption("CreateNamespace=true")
		}
```

Reply:
The option shouldn't be hardcoded, it should use the value that is passed in to the function.